commit 8749291dae4536e44a137234e8db172874603bbb
Author: Trent Mick <trentm@gmail.com>
Date:   2019-07-25T10:52:32-07:00 (2 months ago)
    
    TRITON-1828 drop firmware-tools and "headnode-joyent" flavour headnode build

diff --git a/Makefile b/Makefile
index 1fef5a6c..044a3790 100644
--- a/Makefile
+++ b/Makefile
@@ -12,11 +12,8 @@
 PERCENT := %
 
 #
-# The headnode build has the following variants, declared by the
-# $(HEADNODE_VARIANT) macro:
-# 'debug'           use a debug platform image
-# 'joyent'          include specific firmware for Joyent deployments
-# 'joyent-debug'    a combination of the above
+# We build with non-debug platform bits, unless HEADNODE_VARIANT is set to
+# 'debug'.
 #
 ifdef HEADNODE_VARIANT
     HEADNODE_VARIANT_SUFFIX=-$(HEADNODE_VARIANT)
@@ -28,19 +25,6 @@ ifeq ($(HEADNODE_VARIANT), debug)
     DEBUG_BUILD=true
 endif
 
-ifeq ($(HEADNODE_VARIANT), joyent)
-    JOYENT_BUILD=true
-    # this is an internal build.
-    ENGBLD_DEST_OUT_PATH ?= /stor/builds
-endif
-
-ifeq ($(HEADNODE_VARIANT), joyent-debug)
-    JOYENT_BUILD=true
-    DEBUG_BUILD=true
-    # this is an internal build.
-    ENGBLD_DEST_OUT_PATH ?= /stor/builds
-endif
-
 ifdef DEBUG_BUILD
     DEBUG_SUFFIX=-debug
 endif
@@ -307,20 +291,20 @@ deps: 0-npm-stamp clean-img-cruft build.spec.merged
 .PHONY: coal
 coal: deps download $(TOOLS_DEPS)
 	TIMESTAMP=$(TIMESTAMP) \
-	DEBUG_BUILD=$(DEBUG_BUILD) \
-	JOYENT_BUILD=$(JOYENT_BUILD) bin/build-image coal
+	    DEBUG_BUILD=$(DEBUG_BUILD) \
+	    bin/build-image coal
 
 .PHONY: usb
 usb: deps download $(TOOLS_DEPS)
 	TIMESTAMP=$(TIMESTAMP) \
-	DEBUG_BUILD=$(DEBUG_BUILD) \
-	JOYENT_BUILD=$(JOYENT_BUILD) bin/build-image usb
+	    DEBUG_BUILD=$(DEBUG_BUILD) \
+	    bin/build-image usb
 
 .PHONY: boot
 boot: deps download $(TOOLS_DEPS)
 	TIMESTAMP=$(TIMESTAMP) \
-	DEBUG_BUILD=$(DEBUG_BUILD) \
-	JOYENT_BUILD=$(JOYENT_BUILD) bin/build-image tar
+	    DEBUG_BUILD=$(DEBUG_BUILD) \
+	    bin/build-image tar
 
 .PHONY: tar
 tar: boot
@@ -336,7 +320,6 @@ download: deps
 	$(CHECKER)
 	if [ -z $${NO_DOWNLOAD} ]; then \
 		DEBUG_BUILD=$(DEBUG_BUILD) \
-		JOYENT_BUILD=$(JOYENT_BUILD) \
 		    $(DOWNLOADER) -d -w "log/artefacts.json"; \
 	else \
 		true; \
@@ -620,4 +603,3 @@ bits-upload-latest: build.spec.merged
 
 include ./deps/eng/tools/mk/Makefile.deps
 include ./deps/eng/tools/mk/Makefile.targ
-
diff --git a/README.md b/README.md
index e69f0627..b88aeb2e 100644
--- a/README.md
+++ b/README.md
@@ -259,38 +259,36 @@ subsequent phases of the build:
 
 - `file.agents.sh`
 
+
 By default, the `"manta-base-path"` top-level key is used to specify the
 base directory where the downloader will look for build artefacts in Manta.
 The default value for this key, as shipped in this repository, is
 `"/Joyent_Dev/public/builds"`.  If you wish to include an artefact that
 comes from a different Manta directory tree, you may specify the name of
-an alternative top-level `build.spec` key on a per-file basis.
+an alternative top-level `build.spec` key on a per-file basis via the following:
 
-For example, Joyent ships firmware files for specific server hardware that are
-not available under an opensource license.  As a result, these files are only
-included in the commercially supported builds of Triton to Joyent customers.
-The firmware artefact is stored in a different (Joyent-private) area of Manta,
-and configured thus:
+1. Add a key with the value of the alternate Manta dir, e.g.:
 
-```
-{
-    ...
-    "joyent-manta-base-path": "/Joyent_Dev/stor/builds",
-    ...
-    "files": {
-        "firmware-tools": {
-            "alt_manta_base": "joyent-manta-base-path",
-            "file": { "base": "firmware-tools", "ext": "tgz" }
+    ```
+        "my-private-manta-base": "/mymantauser/stor/builds"
+    ```
+
+2. Specify `"alt_manta_base": "<that added key name>"` in the options for
+   that file, e.g.:
+
+    ```
+        "files": {
+            "sdcadm": {
+                "alt_manta_base": "my-private-manta-base",
+                "file": { "base": "sdcadm", "ext": "sh" }
+            },
+            ...
         },
-        ...
-    },
-    ...
-}
-```
+    ```
+
+    This tells the download phase to use your `my-private-manta-base` path
+    for this artefact.
 
-The `"alt_manta_base"` key specifies that the download phase of the build
-should look in the path specified in `"joyent-manta-base-path"` for this
-artefact, rather than the default key of `"manta-base-path"`.
 
 #### Alternative Branch Selection
 
diff --git a/bin/build-tar-image b/bin/build-tar-image
index abc7c426..44d7ac37 100755
--- a/bin/build-tar-image
+++ b/bin/build-tar-image
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright 2019, Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 if [[ -n "$TRACE" ]]; then
@@ -226,20 +226,6 @@ function copy_base
         plat_suf=""
     fi
 
-    local is_joyent
-    is_joyent="$(${BUILDSPEC} -f joyent-build)"
-
-    if [[ "${is_joyent}" == "true" ]]; then
-        local ftbranch
-        ftbranch=$(build_spec firmware-tools-release)
-        [[ -z ${ftbranch} ]] && ftbranch="master"
-    fi
-
-    if [[ "${is_joyent}" == "true" ]]; then
-        local firmware_path
-        firmware_path=$(get_bit "file.firmware-tools.tgz")
-    fi
-
     echo "==> Creating .joyliveusb file"
     touch ${STAGE}/.joyliveusb
 
@@ -260,13 +246,6 @@ function copy_base
 
     echo "==> Copying in LICENSE"
     cp -r LICENSE ${STAGE}/LICENSE
-
-    if [[ "${is_joyent}" == "true" ]]; then
-        echo "==> Extracting firmware bundle"
-        if ! (cd ${STAGE} && ${TAR} -xzf ${firmware_path}); then
-            fatal "Failed to extract firmware bundle"
-        fi
-    fi
 }
 
 function copy_config
diff --git a/bin/reflash b/bin/reflash
index 75f018c6..9409224e 100755
--- a/bin/reflash
+++ b/bin/reflash
@@ -356,39 +356,24 @@ get_url_content_md5()
     echo "${content_md5}"
 }
 
-get_manta_base_path()
-{
-    local manta_base_path
-
-    manta_base_path=$(build_spec "joyent-manta-base-path")
-
-    if [[ -z ${manta_base_path} ]]; then
-        fatal "Unable to find manta_base_path, check that " \
-            "joyent-manta-base-path is set in your build.spec or " \
-            "build.spec.local"
-    fi
-
-    echo "${manta_base_path}"
-}
-
 get_manta_latest_release()
 {
     local latest_dir
     local release
-    local manta_base
+    local manta_base_path
     local manta_latest_path
     local manta_subdir
 
     latest_dir=
     release=$1
-    manta_base=$(get_manta_base_path)
+    manta_base_path=$(get_build_spec_var "manta-base-path")
     manta_latest_path=
-    manta_subdir="headnode-joyent"
+    manta_subdir="headnode"
 
     [[ -n ${release} ]] || release="master"
     [[ ${release} == "latest" ]] && release="master"
 
-    manta_latest_path="${manta_base}/${manta_subdir}/${release}-latest"
+    manta_latest_path="${manta_base_path}/${manta_subdir}/${release}-latest"
     latest_dir=$(mget_stdout ${manta_latest_path})
 
     [[ -n ${latest_dir} ]] \
diff --git a/build.spec b/build.spec
index cce05036..70a4fc85 100644
--- a/build.spec
+++ b/build.spec
@@ -6,24 +6,17 @@
   "no-rabbit": true,
   "clean-cache": true,
   "smt_enabled": true,
-  "// joyent-build": "set to true to enable ancillary repository use",
 
   "features": {
     "debug-platform": {
       "enabled": false,
       "env": "DEBUG_BUILD"
-    },
-    "joyent-build": {
-      "enabled": false,
-      "env": "JOYENT_BUILD"
     }
   },
 
   "bits-branch": "master",
 
-  "// manta-*": "You can override these in your build.spec.local.",
   "manta-base-path": "/Joyent_Dev/public/builds",
-  "joyent-manta-base-path": "/Joyent_Dev/stor/builds",
   "manta-user": "Joyent_Dev",
   "manta-url": "https://us-east.manta.joyent.com",
 
@@ -102,12 +95,6 @@
     "agents_md5": {
       "jobname": "agentsshar",
       "file": { "base": "agents", "ext": "md5sum" }
-    },
-
-    "firmware-tools": {
-      "if_feature": "joyent-build",
-      "alt_manta_base": "joyent-manta-base-path",
-      "file": { "base": "firmware-tools", "ext": "tgz" }
     }
   }
 }
diff --git a/buildtools/novus/cmd/checker.js b/buildtools/novus/cmd/checker.js
index 2fa3d4fd..f12d1dbc 100644
--- a/buildtools/novus/cmd/checker.js
+++ b/buildtools/novus/cmd/checker.js
@@ -106,7 +106,6 @@ check_old_branch_keys()
 		'platform-image': '"files.platform.*"',
 		'platform-release': '"files.platform.*"',
 		'ipxe-release': '"files.ipxe.*"',
-		'firmware-tools-release': '"files.firmware-tools.*"',
 		'sdcadm-release': '"files.sdcadm.*"',
 		'agents-shar': '"files.agents.*" and "files.agents_md5.*"'
 	};
@@ -136,8 +135,7 @@ function
 check_features()
 {
 	var OLD_TO_NEW = {
-		'debug-platform': '"features.debug-platform.*" or $DEBUG_BUILD',
-		'joyent-build': '"features.joyent-build.*" or $JOYENT_BUILD'
+		'debug-platform': '"features.debug-platform.*" or $DEBUG_BUILD'
 	};
 
 	var lines = [];
